### PR TITLE
[CALCITE-3606] Derive target table column list by mistake when conver…

### PR DESCRIPTION
**Pull request by jinxing64**
…t TableModify to Sql string (JinXing)

We should derive the target table column list by the target table itself, not from the input.

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java [hunk 1]
```diff
@@ -702,7 +702,7 @@ public Result visit(TableModify modify) {
 
       final SqlInsert sqlInsert =
           new SqlInsert(POS, SqlNodeList.EMPTY, sqlTargetTable, sqlSource,
-              identifierList(modify.getInput().getRowType().getFieldNames()));
+              identifierList(modify.getTable().getRowType().getFieldNames()));
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: Shouldn't we use `TableModify.getUpdateColumnList` ? Also add a test for insert with explicit table columns.

Comment (jinxing64) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: Thanks, Danny~
I will refine by your comments soon.

Comment (jinxing64) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: `TableModify.getUpdateColumnList` returns null for `Operation.INSERT` [1]
I added a test for insert with explicit table columns.

[1] https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rel/core/TableModify.java#L125

**Comment by jinxing64** [comment]
Thanks for approval, Danny and Chunwei ~

**Pull request closed without merging**